salt/modules/reg.py

```diff
@@ -142,6 +142,54 @@ def broadcast_change():
     return not bool(res)
 
 
+def list_keys(hive, key=None, use_32bit_registry=False):
+    '''
+    Enumerates the subkeys in a registry key or hive.
+
+    :param str hive: The name of the hive. Can be one of the following
+
+        - HKEY_LOCAL_MACHINE or HKLM
+        - HKEY_CURRENT_USER or HKCU
+        - HKEY_USER or HKU
+
+    :param str key: The key (looks like a path) to the value name. If a key is
+        not passed, the keys under the hive will be returned.
+
+    :param bool use_32bit_registry: Accesses the 32bit portion of the registry
+        on 64 bit installations. On 32bit machines this is ignored.
+
+    :return: A list of keys/subkeys under the hive or key.
+    :rtype: list
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' reg.list_keys HKLM 'SOFTWARE'
+    '''
+    registry = Registry()
+    hkey = registry.hkeys[hive]
+    access_mask = registry.registry_32[use_32bit_registry]
+
+    subkeys = []
+    try:
+        handle = _winreg.OpenKey(hkey, key, 0, access_mask)
+        i = 0
+        while True:
+            try:
+                subkey = _winreg.EnumKey(handle, i)
+                subkeys.append(subkey)
+                i += 1
+            except WindowsError:  # pylint: disable=E0602
```

Comment:
This is a little scary. Could we make this a little safer? I'm concerned that this might result in an infinite loop and bring down a machine.